### DISCUSSION TITLE
Model transformation.

### DISCUSSION THREAD

**Discussion by boykan313** [SYSTRAN/faster-whisper#317] [Q&A]
Hello! I have a problem converting the whisper model to faster-whisper. After this process, I don’t understand what to do, the process is very difficult. Help me please.
I am using this model: https://huggingface.co/mitchelldehaven/whisper-large-v2-ru
Transformed through this command: ct2-transformers-converter --model mitchelldehaven/whisper-large-v2-ru --output_dir whisper-large-v2-ru-ct2 --copy_files tokenizer.json --quantization float16
![322221](https://github.com/guillaumekln/faster-whisper/assets/90963133/b469261e-2500-46c8-915c-2bf04f357494)

**Comment by boykan313** [comment]
Now I'm learning such a result when transforming the model.
![343](https://github.com/guillaumekln/faster-whisper/assets/90963133/03ee35b7-9e90-4a7e-82ae-d676910c5fbb)

**Comment by guillaumekln** [comment]
Try converting without `--copy_files tokenizer.json`.

**Reply by boykan313** [reply]
![32114](https://github.com/guillaumekln/faster-whisper/assets/90963133/eb75b384-0d6d-4285-a224-e7cf627c39c0)

Nothing happens. It is heard that the computer accelerates for a couple of seconds and no result.

**Reply by boykan313** [reply]
![33333](https://github.com/guillaumekln/faster-whisper/assets/90963133/4be071f2-387d-4639-bb27-821ccf4dff7b)
entered this command again, got the following output.

**Reply by boykan313** [reply]
![4444](https://github.com/guillaumekln/faster-whisper/assets/90963133/81bd7cd2-bf33-490c-a2b5-e1d18c2ca3ad)
eventually came to this situation. Nothing happens.

**Reply by boykan313** [reply]
![5555](https://github.com/guillaumekln/faster-whisper/assets/90963133/67cf2162-0a7f-4790-90c3-e3a2375cdee3)
![11111](https://github.com/guillaumekln/faster-whisper/assets/90963133/9731cf34-2a88-41a9-82d4-e1897aa92d52)
It can be seen that some process is going on, but there is no result.

**Reply by guillaumekln** [reply]
The script created the directory `whisper-large-v2-ru-ct2` so the conversion worked.

**Reply by boykan313** [reply]
I already understood this... I found the directory of this model, renamed it to large-v2 and launched it, downloaded one file and started the process. it seems to work ... I thought this model would be read as a separate one, but in the end I had to crush the existing model of this one, under the same name.
this process was so primitive?

**Reply by boykan313** [reply]
what you said worked only after the complete removal of the downloaded model, but without the command that you called. I'm glad it all worked out in the end. Thank you for your help and experience! Don't abandon this project!

**Reply by boykan313** [reply]
Tell me please, is it possible to somehow spoil the model during education? My new model, which is aimed at the Russian language, works worse than your universal one.

**Reply by boykan313** [reply]
Sorry for bothering you often, can you help me with this problem with my model? She does not want to start, whisper does not see her.

**Reply by guillaumekln** [reply]
It seems you are using https://github.com/Purfview/whisper-standalone-win to run the model, so you should ask questions in this other project. Your latest error has nothing to do with faster-whisper.

**Comment by Purfview** [comment]
> I already understood this... I found the directory of this model, renamed it to large-v2 and launched it, downloaded one file and started the process. it seems to work ... I thought this model would be read as a separate one, but in the end I had to crush the existing model of this one, under the same name. this process was so primitive?

Don't replace existing model, just run converted model from new folder.

**Reply by boykan313** [reply]
figured it out! The name after the model conversion did not completely change the name of the folder, it was just a whisper, added faster and it all worked! Thank you for your patience and help! Develop the project further, I will follow the updates!

**Comment by Purfview** [comment]
> Tell me please, is it possible to somehow spoil the model during education? My new model, which is aimed at the Russian language, works worse than your universal one.

It's possible.

**Reply by boykan313** [reply]
Is it possible to remove this effect when converting?

**Reply by Purfview** [reply]
No.

**Reply by boykan313** [reply]
![Screenshot_1](https://github.com/guillaumekln/faster-whisper/assets/90963133/ee62865c-6f16-4d5a-ad6f-d4d0665d513e)
Can you help me? Why doesn't whisper see the new model?

**Reply by Purfview** [reply]
Your model is named `medium-ru-ct2` when your command is for non-existant `model-ru-ct2`.

**Reply by boykan313** [reply]
this is an old screenshot, sorry, I noticed it after posting. Changing the title does not change anything.

**Reply by boykan313** [reply]
![Screenshot_3](https://github.com/guillaumekln/faster-whisper/assets/90963133/b7887ba2-d2a0-471d-9cbe-8a59b605c042)

**Reply by Purfview** [reply]
You wrote that you have missing "tokenizer.json" file, that's why this error is produced by Standalone Faster-Whisper.

**Reply by boykan313** [reply]
Yes, I figured that out too, thanks. This happened when the model was transformed, I asked a question about the author. I want to ask you if there is a difference in running the converted model through your work and the original faster whispr through python?

**Reply by boykan313** [reply]
Maybe I asked a stupid question ... It's just that the conversion instructions say that when the model starts, it should download this file.

**Reply by Purfview** [reply]
In Standalone Faster-Whisper download works only if you use the original models,

**Reply by boykan313** [reply]
So the problem came to light ... but can it be fixed somehow or will support be added in the future? It would be very useful in my work...

**Reply by Purfview** [reply]
Useful what?

**Reply by boykan313** [reply]
If it will be possible to run third party converted models for faster-whisper.

**Reply by Purfview** [reply]
What prevents you running them now?

**Reply by boykan313** [reply]
I'm sorry, I first wrote and then I thought. He already supports it, but he cannot download the file. Sorry.

**Reply by Purfview** [reply]
Not sure what you expect to be downloaded if a custom model is on your computer already.
Copy `tokenizer.json` from some original model. [it's same in all original models]

**Reply by boykan313** [reply]
the absence of a file that should be downloaded after starting the model, according to this instruction.
https://github.com/guillaumekln/faster-whisper#model-conversion

**Reply by boykan313** [reply]
I did that and the transcription was terrible.

**Reply by Purfview** [reply]
Downloading same file wouldn't change your results.

**Reply by boykan313** [reply]
It turns out that this file is the same for all models? there is only a binding to medium, large, etc.?

**Reply by Purfview** [reply]
Yes.

**Reply by boykan313** [reply]
which is strange, running it in a normal whisker, it worked, but for some reason only a couple of seconds, then it freezes but the processor still loads.

**Comment by boykan313** [comment]
> No.



> The script created the directory `whisper-large-v2-ru-ct2` so the conversion worked.



> The script created the directory `whisper-large-v2-ru-ct2` so the conversion worked.



> The script created the directory `whisper-large-v2-ru-ct2` so the conversion worked.

As I understand it, whisper does not see the model, because it does not contain the "tokenizer.json" file. In the instructions for transforming the model, it was said that this file should be downloaded later on the next start, but this does not happen. You can copy this file from the original large-v2 model and then the new model will be detected, but it will not work correctly..

**Reply by boykan313** [reply]
Sorry, I want to ask about the model transformation process. The transformation was successful but without the "tokenizer.json" file. Where can I get it now so that the model works? Your instructions say that it will download itself after the next launch, but this did not happen.

**Reply by Purfview** [reply]
I don't know much about the model transformation process nor if "tokenizer.json" file is mandatory.
Currently Standalone Faster-Whisper. checks if all 4 files are present, same named files as in guillaumekln's original models. 

EDIT:
Try to copy `tokenizer.json` from some original model. [it's same in all original models]
